frida/core.py

```diff
@@ -358,11 +417,41 @@ def list_exports(self) -> List[str]:
         assert isinstance(result, list)
         return result
 
+    def list_exports(self) -> List[str]:
+        """
+        List all the exported attributes from the script's rpc
+        """
+
+        warnings.warn(
+            "Script.list_exports will become asynchronous in the future, use the explicit Script.list_exports_sync instead",
+            DeprecationWarning,
+        )
+        return self.list_exports_sync()
+
+    def _rpc_request_async(self, *args: Any) -> asyncio.Future[Any]:
```

Comment:
Generic `asyncio.Future` is only available in python 3.9 and later :(
I can think of 3 possible solutions:

1. Only define the generic during type checking, like this:
```python
if TYPE_CHECKING:
    _AnyFuture = asyncio.Future[Any]
else:
    _AnyFuture = asyncio.Future
```
2.  Use `from __future__ import annotations`
3.  Don't use it as a generic

Reply:
I've added the annotations import, but I don't think it will help, will it? It will just postpone the error to when we call the method

Reply:
No, the import will make python treat annotations as strings, and will not evaluate them at all during runtime.

You can try it on https://www.online-python.com/k8azfliZIV which uses an older python version